### PR TITLE
Scan recommendations

### PR DESCRIPTION
**Pull request by RoiGlinik**

### CODE REVIEW

File: playbooks/robusta_playbooks/popeye.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    start_time = end_time = datetime.now()
+    popeye_scan = scan = {} 
+    logs = None
+    
+    try:
+        logs = RobustaJob.run_simple_job_spec(spec,"popeye_job",params.timeout)
+        scan = json.loads(logs)
+        end_time = datetime.now() 
+        popeye_scan = PopeyeReport(**scan['popeye'])
+    except JSONDecodeError as e:
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/popeye.py:150]: I'm not sure we should create a finding each time the scan fails.
Might be ok in the early stages of the feature

File: playbooks/robusta_playbooks/popeye.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        end_time=end_time,
+        score=popeye_scan.score,
+        results=[],
+        config=f"{params.args} \n\n {params.spinach}",
+        pdf_scan_row_content_format=scanRowContentToString,
+        pdf_scan_row_priority_format=levelToString
+        )
+
+    scan_issues: List[ScanReportRow] = []
+    for section in popeye_scan.sanitizers:
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/popeye.py:178]: Can `sanitizers` be None? (it's Optional)

File: playbooks/robusta_playbooks/popeye.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        kind = section.sanitizer
+        issuesDict: Dict[str,List[Issue]] = section.issues or {}
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/popeye.py:180]: snake case

File: playbooks/robusta_playbooks/popeye.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        for resource, issuesList  in issuesDict.items():
+            namespace, _ , name = resource.rpartition("/")
+
+            groupedIssues = group_issues_list(issuesList)
```
Comment (arikalon1) [inline playbooks/robusta_playbooks/popeye.py:184]: snake case

File: src/robusta/core/reporting/blocks.py [hunk 5]
```diff
@@ -386,3 +389,42 @@ def __init__(self, data: PrometheusQueryResult, query: str):
         """
         metadata = {"query-result-version": "1.0", "query": query}
         super().__init__(data=data, metadata=metadata)
+
+
+class ScanReportRow(BaseModel):
+    scan_id: str  # UUID
+    namespace: str
+    scan_type: ScanType
```
Comment (arikalon1) [inline src/robusta/core/reporting/blocks.py:397]: Are all the fields mandatory?

File: src/robusta/core/sinks/robusta/dal/supabase_dal.py [hunk 6]
```diff
@@ -96,27 +104,79 @@ def __init__(
...
+    def __to_db_scanResult(self, scanResult: ScanReportRow) -> Dict[Any, Any]:
+        db_sr = scanResult.dict()
+        db_sr["account_id"] = self.account_id
+        db_sr["cluster_id"] = self.cluster
+        return db_sr
+    
+
+    def persist_scan(self, enrichment: Enrichment):
+
+        for block in enrichment.blocks:
```
Comment (arikalon1) [inline src/robusta/core/sinks/robusta/dal/supabase_dal.py:118]: Would it be better to persist all scan report blocks?
like:
```
for block in emrichments.blocks if isinstance(block, ScanReportBlock):
  # persist scan
```
We're not supporting the case there are multiple blocks in the enrichment
Is it too restrictive ?

File: src/robusta/core/sinks/robusta/dal/supabase_dal.py [hunk 7]
```diff
@@ -96,27 +104,79 @@ def __init__(
...
+        db_scanResults = [self.__to_db_scanResult(sr) for sr in block.results]
+        res = self.client.table(SCANS_RESULT_TABLE).insert(db_scanResults).execute()
+        if res.get("status_code") not in [200, 201]:
+            msg = f"Failed to persist scan {block.scan_id} error: {res.get('data')}"
+            logging.error(msg)
+            self.handle_supabase_error()
+            raise Exception(msg)
+
+        self.client.postgrest.session.headers.update(self.client._get_auth_headers())
+        tasks = [self.event_loop.create_task(self.client.rpc("insert_scan_meta", {
```
Comment (arikalon1) [inline src/robusta/core/sinks/robusta/dal/supabase_dal.py:131]: we added recently a sync rpc call impl
https://github.com/robusta-dev/robusta/blob/1d366f882cbe064ef70da4f5f2bad9ead7351cd6/src/robusta/core/sinks/robusta/dal/supabase_dal.py#L335

I think it's simpler and safer to use

File: src/robusta/core/sinks/robusta/dal/supabase_dal.py [hunk 8]
```diff
@@ -96,27 +104,79 @@ def __init__(
...
-                    ModelConversion.to_evidence_json(
-                        account_id=self.account_id,
-                        cluster_id=self.cluster,
-                        sink_name=self.sink_name,
-                        signing_key=self.signing_key,
-                        finding_id=finding.id,
-                        enrichment=enrichment,
+
+        scans, enrichments = [], []
+        for e in finding.enrichments:
```
Comment (arikalon1) [inline src/robusta/core/sinks/robusta/dal/supabase_dal.py:152]: please call it enrich or something. e is exception

File: src/robusta/core/sinks/robusta/dal/supabase_dal.py [hunk 9]
```diff
@@ -96,27 +104,79 @@ def __init__(
...
+            scans.append(e) if e.annotations.get(EnrichmentAnnotation.SCAN, False) else enrichments.append(e)
+
+        for s in scans:
```
Comment (arikalon1) [inline src/robusta/core/sinks/robusta/dal/supabase_dal.py:155]: and scan

File: src/robusta/core/sinks/robusta/robusta_sink.py [hunk 10]
```diff
@@ -54,6 +55,7 @@ def __init__(self, sink_config: RobustaSinkConfigWrapper, registry):
             sink_config.robusta_sink.name,
             self.cluster_name,
             self.signing_key,
+            asyncio.get_event_loop()
```
Comment (arikalon1) [inline src/robusta/core/sinks/robusta/robusta_sink.py:58]: I think we don't need this

File: src/robusta/core/sinks/transformer.py [hunk 11]
```diff
@@ -171,3 +175,87 @@ def tableblock_to_fileblocks(blocks: List[BaseBlock], column_limit: int) -> List
...
+    @staticmethod
+    def scanReportBlock_to_fileblock(block: BaseBlock) -> BaseBlock:
+
+        if not isinstance(block, ScanReportBlock):
+            return block
+        
+        accent_color = (140, 249, 209)
+        headers_color = (63, 63, 63)
+        table_color = (207, 215, 216)
+        def setNormalTextStyle(pdf: FPDF):
```
Comment (arikalon1) [inline src/robusta/core/sinks/transformer.py:189]: I would change functions to snake casee

**Review by arikalon1 — APPROVED** [review]
very nice work
Left some comments

File: helm/robusta/values.yaml [hunk 12]
```diff
@@ -303,6 +304,15 @@ platformPlaybooks:
   - job_pod_enricher: {}
   sinks:
     - "robusta_ui_sink"
+- triggers:
```
[the remark this answers is not in the document]
Comment (RoiGlinik) [reply helm/robusta/values.yaml:307]: You are right, will change it to run once on startup

File: src/robusta/core/sinks/robusta/dal/supabase_dal.py (hunk 6 above)
Comment (RoiGlinik) [reply src/robusta/core/sinks/robusta/dal/supabase_dal.py:118]: yea , was a mistake

**Pull request merged**